examples/image_segmentation/android/app/src/main/java/com/google/mediapipe/examples/imagesegmenter/fragments/CameraFragment.kt

```diff
@@ -0,0 +1,276 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.Toast
+import androidx.camera.core.AspectRatio
+import androidx.camera.core.Camera
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.Navigation
+import com.google.mediapipe.examples.imagesegmenter.ImageSegmenterHelper
+import com.google.mediapipe.examples.imagesegmenter.MainViewModel
+import com.google.mediapipe.examples.imagesegmenter.R
+import com.google.mediapipe.examples.imagesegmenter.databinding.FragmentCameraBinding
+import com.google.mediapipe.tasks.vision.core.RunningMode
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+class CameraFragment : Fragment(), ImageSegmenterHelper.SegmenterListener {
+
+    companion object {
+        private const val TAG = "Image segmenter"
+    }
+
+    private var _fragmentCameraBinding: FragmentCameraBinding? = null
+    private val fragmentCameraBinding
+        get() = _fragmentCameraBinding!!
+
+    private val viewModel: MainViewModel by activityViewModels()
+    private lateinit var imageSegmenterHelper: ImageSegmenterHelper
+    private var preview: Preview? = null
+    private var imageAnalyzer: ImageAnalysis? = null
+    private var camera: Camera? = null
+    private var cameraProvider: ProcessCameraProvider? = null
+
+    /** Blocking operations are performed using this executor */
+    private var backgroundExecutor: ExecutorService? = null
+
+
+    override fun onResume() {
+        super.onResume()
+
+        if (!PermissionsFragment.hasPermissions(requireContext())) {
+            Navigation.findNavController(
+                requireActivity(), R.id.fragment_container
+            ).navigate(CameraFragmentDirections.actionCameraToPermissions())
+        }
+
+        backgroundExecutor?.execute {
+            if (imageSegmenterHelper.isClosed()) {
+                imageSegmenterHelper.setListener(this)
+                imageSegmenterHelper.setupImageSegmenter()
+            }
+        }
+    }
+
+    override fun onPause() {
+        // save ImageSegmenter settings
+        viewModel.setDelegate(imageSegmenterHelper.currentDelegate)
+        super.onPause()
+
+        // Close the image segmenter and release resources
+        backgroundExecutor?.execute {
+            with(imageSegmenterHelper) {
+                clearListener()
+                clearImageSegmenter()
+            }
+        }
+    }
+
+    override fun onDestroyView() {
+        _fragmentCameraBinding = null
+
+        // Shut down our background executor
+        backgroundExecutor?.shutdown()
+        backgroundExecutor?.awaitTermination(
```

Comment:
Why are we using await for terminating the background executor here after shutdown?

Reply:
Because follow the docs.
`Shutdown: previously submitted tasks are executed, but no new tasks will be accepted.`
and 
`awaitTermination: Blocks until all tasks have completed execution after a shutdown request, or the timeout occurs, or the current thread is interrupted, whichever happens first.`
That why i call shutdown() before awaitTermination()

Reply:
Ah cool, OK.